p2p/src/disconnection/p2p_disconnection_actions.rs

```diff
@@ -21,8 +21,16 @@ pub enum P2pDisconnectionAction {
 impl redux::EnablingCondition<P2pState> for P2pDisconnectionAction {
     fn is_enabled(&self, state: &P2pState, _time: redux::Timestamp) -> bool {
         match self {
-            P2pDisconnectionAction::Init { peer_id, .. }
-            | P2pDisconnectionAction::Finish { peer_id } => {
+            P2pDisconnectionAction::Init { peer_id, .. } => {
+                state.peers.get(peer_id).map_or(false, |peer| {
+                    !matches!(peer.status, P2pPeerStatus::Disconnected { .. })
+                }) && state.network.scheduler.connections.iter().any(|(_, st)| {
```

Comment:
You are expecting all peers to be libp2p, but webrtc peers won't have libp2p state so it will break that.
Why do you need this extra check anyways? If peer is not in disconnected state, but is in some other state, we need to enable this action.

Reply:
Without this check, I see timeout disconnects repeated many times before the actual disconnect happens. Is it better to add a `P2pPeerStatus::Disconnecting` state to indicate that a disconnect is pending?

Reply:
Why does it take that long to disconnect a peer after disconnection was initiated though?

Reply:
It takes microseconds. The problem was that the peer was not marked as disconnected because the connection removal is postponed and [this](https://github.com/openmina/openmina/blob/4b4dadfb2c381cc46205c172a313050636264e0f/p2p/src/disconnection/p2p_disconnection_reducer.rs#L68) check was triggered. I fixed it by the condition `conn_state.peer_id() == Some(peer_id) && conn_state.closed.is_none()`.

Reply:
I removed this from enabling conditions. WebRTC should be ok.